### PR TITLE
[kube-prometheus-stack] Allow to directly config prometheus.spec and  alertmanager.spec

### PR DESCRIPTION
**Pull request by jkroepke**
… alertmanager.spec

#### What this PR does / why we need it

This is useful, if complex templating is required or properties are not exposed by the chart yet.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by GMartinez-Sisti** [comment]
@jkroepke WDYT of adding a couple of values from these new keys to https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/ci/03-non-defaults-values.yaml? This way we can ensure they keep working as expected.

File: charts/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -876,6 +876,14 @@ alertmanager:
     ## be considered available. Defaults to 0 (pod will be considered available as soon as it is ready).
     minReadySeconds: 0
 
+    ## Additional configuration which is not covered by the properties above. (passed through tpl)
+    additionalConfig: {}
+
+    ## Additional configuration which is not covered by the properties above.
+    ## Useful, if you need advanced templating inside alertmanagerSpec.
+    ## Otherwise, use alertmanager.alertmanagerSpec.additionalConfig (passed through tpl)
+    additionalConfigString: {}
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/values.yaml:885]: Thanks @jkroepke, this causes the following output on the lint job:

[1905](https://github.com/prometheus-community/helm-charts/actions/runs/6584932364/job/17890388114?pr=3920#step:9:1906) coalesce.go:220: warning: cannot overwrite table with non table for kube-prometheus-stack.alertmanager.alertmanagerSpec.additionalConfigString (map[])
[1906](https://github.com/prometheus-community/helm-charts/actions/runs/6584932364/job/17890388114?pr=3920#step:9:1907) coalesce.go:220: warning: cannot overwrite table with non table for kube-prometheus-stack.alertmanager.alertmanagerSpec.additionalConfigString (map[])

I think you want this:

```suggestion
    additionalConfigString: ""
```

**Comment by GMartinez-Sisti** [comment]
Did you mean `Allow to directly` instead of `Allow to directory` 👼 ?

**Comment by jkroepke** [comment]
> Did you mean `Allow to directly` instead of `Allow to directory` 👼 ?

GoLand spell correction, yeah. Thanks!

**Comment by jkroepke** [comment]
> @jkroepke WDYT of adding a couple of values from these new keys to [`main`/charts/kube-prometheus-stack/ci/03-non-defaults-values.yaml](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/ci/03-non-defaults-values.yaml?rgh-link-date=2023-10-19T22%3A29%3A04Z)? This way we can ensure they keep working as expected.

I really like this way to testing and we should think about the make it as requirement ...

**Comment by jkroepke** [comment]
> I agree we should use it more, the only downside is that it can get hard to troubleshoot

Internally, we run helm install command with --debug which solves the throubleshoot issues

**Pull request merged**